acceptance/tests/partitions/partitions_sync_test.go

```diff
@@ -0,0 +1,297 @@
+			defaultNamespace,
+			false,
+			true,
+		},
+		{
+			"single destination namespace",
+			staticServerNamespace,
+			false,
+			false,
+		},
+		{
+			"single destination namespace; ACLs and auto-encrypt enabled",
+			staticServerNamespace,
+			false,
+			true,
+		},
+		{
+			"mirror k8s namespaces",
+			staticServerNamespace,
+			true,
+			false,
+		},
+		{
+			"mirror k8s namespaces; ACLs and auto-encrypt enabled",
+			staticServerNamespace,
+			true,
+			true,
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			serverClusterContext := env.DefaultContext(t)
+			clientClusterContext := env.Context(t, environment.SecondaryContextName)
+
+			ctx := context.Background()
+
+			commonHelmValues := map[string]string{
+				"global.adminPartitions.enabled": "true",
+
+				"global.enableConsulNamespaces": "true",
+
+				"global.tls.enabled":           "true",
+				"global.tls.httpsOnly":         strconv.FormatBool(c.ACLsAndAutoEncryptEnabled),
+				"global.tls.enableAutoEncrypt": strconv.FormatBool(c.ACLsAndAutoEncryptEnabled),
+
+				"global.acls.manageSystemACLs": strconv.FormatBool(c.ACLsAndAutoEncryptEnabled),
+
+				"syncCatalog.enabled": "true",
+				// When mirroringK8S is set, this setting is ignored.
+				"syncCatalog.consulNamespaces.consulDestinationNamespace": c.destinationNamespace,
+				"syncCatalog.consulNamespaces.mirroringK8S":               strconv.FormatBool(c.mirrorK8S),
+				"syncCatalog.addK8SNamespaceSuffix":                       "false",
+
+				"controller.enabled": "true",
+
+				"dns.enabled":           "true",
+				"dns.enableRedirection": strconv.FormatBool(cfg.EnableTransparentProxy),
+			}
+
+			serverHelmValues := map[string]string{
+				"server.exposeGossipAndRPCPorts": "true",
+			}
+
+			// On Kind, there are no load balancers but since all clusters
+			// share the same node network (docker bridge), we can use
+			// a NodePort service so that we can access node(s) in a different Kind cluster.
+			if cfg.UseKind {
+				serverHelmValues["global.adminPartitions.service.type"] = "NodePort"
+				serverHelmValues["global.adminPartitions.service.nodePort.https"] = "30000"
+			}
+
+			releaseName := helpers.RandomName()
+
+			helpers.MergeMaps(serverHelmValues, commonHelmValues)
+
+			// Install the consul cluster with servers in the default kubernetes context.
+			serverConsulCluster := consul.NewHelmCluster(t, serverHelmValues, serverClusterContext, cfg, releaseName)
+			serverConsulCluster.Create(t)
+
+			// Get the TLS CA certificate and key secret from the server cluster and apply it to the client cluster.
+			caCertSecretName := fmt.Sprintf("%s-consul-ca-cert", releaseName)
+			caKeySecretName := fmt.Sprintf("%s-consul-ca-key", releaseName)
+
+			logger.Logf(t, "retrieving ca cert secret %s from the server cluster and applying to the client cluster", caCertSecretName)
+			moveSecret(t, serverClusterContext, clientClusterContext, caCertSecretName)
```

Comment:
I know you didn't add this function, but I'm wondering if it might be clearer if it were called `copySecret`. "Move" makes me think that the original instance is no longer there.